### PR TITLE
Close #202: new flag visual_studio_code_gpgcheck

### LINKED ISSUE #202: Add possibility to mark the microsoft repo gpg key as trusted
Sometimes it is needed to mark the microsoft ssl certificate as trusted. for example when you are behind a company proxy and this proxy changes the certificate a bit.

in APT you can mark the repo cert as trusted with `trusted=yes`
example: `deb [arch=amd64 trusted=yes] https://packages.microsoft.com//repos/code stable main`
source list documentation: https://manpages.debian.org/jessie/apt/sources.list.5.en.html
> trusted=yes can be set to indicate that packages from this source are always authenticated even if the Release file is not signed or the signature can't be checked. This disables parts of apt-secure(8) and should therefore only be used in a local and trusted context. trusted=no is the opposite which handles even correctly authenticated sources as not authenticated.

For RPM source files a similar config value is `gpgcheck`
https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux/6/html/deployment_guide/sec-configuring_yum_and_yum_repositories
> `gpgcheck=value`
    where value is one of:
    0 — Disable GPG signature-checking on packages in all repositories, including local package installation.
    1 — Enable GPG signature-checking on all packages in all repositories, including local package installation. gpgcheck=1 is the default, and thus all packages' signatures are checked.

### PR DESCRIPTION
**Pull request by arolfes**
when flag `visual_studio_code_gpgcheck` is set to `yes`
- for apt repo the option trusted=yes is NOT added
- for dnf/yum the option gpgcheck is set to yes
- for zypper the option gpgcheck is set to 1
yes is the default

when flag `visual_studio_code_gpgcheck` is set to `no`
- for apt repo the option trusted=yes is added to repo definition
- for dnf/yum the option gpgcheck is set to no
- for zypper the option gpgcheck is set to 0

### CODE REVIEW

File: tasks/install-apt.yml [hunk 1]
```diff
@@ -21,10 +21,20 @@
...
+- name: register variable trusted=yes for gpgcheck
+  set_fact:
+    trusted: 'trusted=yes'
+  when: not visual_studio_code_gpgcheck
+
 - name: install VS Code repo (apt)
   become: yes
   apt_repository:
-    repo: 'deb [arch=amd64] {{ visual_studio_code_mirror }}/repos/code stable main'
+    repo: 'deb [arch=amd64 {{ trusted }}] {{ visual_studio_code_mirror }}/repos/code stable main'
```
Comment (freemanjp) [inline tasks/install-apt.yml:37]: I'd rather you use the ternary filter e.g.:
```yaml
"deb [arch=amd64 {{ visual_studio_code_gpgcheck | ternary('', 'trusted=yes') }}] {{ visual_studio_code_mirror }}/repos/code stable main"
```

File: templates/vscode.repo.j2 [hunk 2]
```diff
@@ -4,5 +4,5 @@ name=Visual Studio Code
 baseurl={{ visual_studio_code_mirror }}/yumrepos/vscode
 enabled=1
 type=rpm-md
-gpgcheck=1
+gpgcheck={{ gpgcheck_zypper }}
```
Comment (freemanjp) [inline templates/vscode.repo.j2:7]: I'd rather you use the ternary filter e.g.:
```jinja2
gpgcheck={{ visual_studio_code_gpgcheck | ternary('1', '0') }}
```

**Review by freemanjp — CHANGES_REQUESTED** [review]

File: tasks/install-apt.yml (hunk 1 above)
Comment (arolfes) [reply tasks/install-apt.yml]: thank you for hint. I didn't know this feature.

i made a separate commit. i can squash it if you like

File: templates/vscode.repo.j2 [hunk 3]
```diff
@@ -4,5 +4,5 @@ name=Visual Studio Code
 baseurl={{ visual_studio_code_mirror }}/yumrepos/vscode
 enabled=1
 type=rpm-md
-gpgcheck=1
+gpgcheck={{ visual_studio_code_gpgcheck | ternary('', 'trusted=yes') }}
```
Comment (freemanjp) [inline templates/vscode.repo.j2:7]: Copy and paste error, should be:
```
gpgcheck={{ visual_studio_code_gpgcheck | ternary('1', '0') }}
```

File: tasks/install-zypper.yml [hunk 4]
```diff
@@ -16,6 +16,16 @@
     state: present
     key: '{{ visual_studio_code_mirror }}/keys/microsoft.asc'
 
+- name: register variable gpgcheck=1
```
Comment (freemanjp) [inline tasks/install-zypper.yml:19]: Remove these as the ternary makes them redundant.

**Review by freemanjp — CHANGES_REQUESTED** [review]

**Comment by freemanjp** [comment]
@arolfes thanks for the pull-requests. You can find your changes in version [6.8.0](https://github.com/gantsign/ansible-role-visual-studio-code/releases/tag/6.8.0) of this role.

**Pull request merged**